### ISSUE TITLE
Drush site-install leads to potentially escalated database privileges

### ISSUE THREAD

**Issue by aborkowski** [drush-ops/drush#4066]
**Describe the bug**
Unescaped MariaDB/MySQL database names lead to a potential privilege escalation when installing Drupal sites using `drush site-install`.

**To Reproduce**
1. Install one copy of Drupal 7 using any method - including `drush site-install` - with a MariaDB database named 'test1' and a new database user (e.g., 'test1'). 

```sh
drush site-install --db-su=... --db-su-pw=... \
--db-url=mysql://test1:test1@localhost/test1`
```
2. Install another copy of Drupal 7 using `drush site-install` with a MariaDB database on the same host as the one in step 1, but with a database named '_____' (that's five underscores) and a new database user (e.g. 'test2').

```sh
drush site-install --db-su=... --db-su-pw=... \
--db-url=mysql://test2:test2@localhost/_____
```

This also works with fewer or more underscores, or a mixture of underscores and letters as long as the database name in step 1 matches the database name in step 2 when treating an underscore as single letter wildcard.

**Expected behavior**
The test2 database user should only be able to access the _____ database.

**Actual behavior**
The test2 database user can access all databases on the same host which have a name that is five characters long, including the test1 database.

**Workaround**
Manually fix the database permissions afterwards with the properly escaped database name.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 8.2.3
| Drupal version? | 7.67
| PHP version?     | 7.0.33
| MariaDB version? | 10.0.38
| OS?             | Debian GNU/Linux 9.9

**Additional information**
I was not able to find a reference for this in the MariaDB documentation, but [related documentation for MySQL](https://dev.mysql.com/doc/refman/8.0/en/grant.html) states:

> The _ and % wildcards are permitted when specifying database names in GRANT statements that grant privileges at the database level (GRANT ... ON db_name.\*). This means, for example, that to use a _ character as part of a database name, specify it as \\_ in the GRANT statement, to prevent the user from being able to access additional databases matching the wildcard pattern; for example, GRANT ... ON `foo\_bar`.\* TO .... 

Affected source code lines are in [Drush 8.2.3](
https://github.com/drush-ops/drush/blob/bf2b2471b9410e0ff4adc23aee020ee1aff28610/lib/Drush/Sql/Sqlmysql.php#L96) and [Drush 9.6.2](https://github.com/drush-ops/drush/blob/65d36cf542308d0b88f77c80f818a978d2844b80/src/Sql/SqlMysql.php#L103).

**Comment by greg-1-anderson** [comment]
This is certainly interesting behavior, and good to know. However, we generally consider privilege escalation bugs to be out of scope for Drush, because a user who can call Drush can already make equivalent attacks, e.g. by calling `mysql` directly.

**Comment by aborkowski** [comment]
Yes, I considered that. And it's easy enough to work around that for the Drush user if you are aware the issue exists. But wouldn't this behaviour give anyone who can execute PHP code on Drupal site2 access to the "matching" databases? They may not be the same person(s).

**Comment by weitzman** [comment]
It would. in your scenario, you have 2 malicious actors. The one who sets up test2, and the one with `Execute PHP` perms. This is beyond reasonable, so I consider this a minor bug.
